torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -348,3 +360,119 @@ def _recurse_update_dict(module, destination, prefix):
     for submodule_name, submodule in module.named_modules():
         if submodule_name != '':
             _recurse_update_dict(submodule, destination, prefix + submodule_name + '.')
+
+def shard_parameter(
+        module: torch.nn.Module,
+        param_name: str,
+        sharding_spec: ShardingSpec,
```

Comment:
I guess implicitly we require module and sharding_spec the same across all ranks ? 
- does it make sense to have some precheck to ensure they are indeed the same before sharding?